routing/index_router.cpp

```diff
@@ -493,9 +493,14 @@ IRouter::ResultCode IndexRouter::CalculateSubroute(Checkpoints const & checkpoin
     return isLastSubroute ? IRouter::EndPointNotFound : IRouter::IntermediatePointNotFound;
   }
 
-  graph.SetMode(AreMwmsNear(startSegment.GetMwmId(), finishSegment.GetMwmId())
-                    ? WorldGraph::Mode::LeapsIfPossible
-                    : WorldGraph::Mode::LeapsOnly);
+  // We use leaps for cars only. Other vehicle types do not have weights in their cross-mwm sections.
+  if (m_vehicleType != VehicleType::Car)
```

Comment:
I suggest you to use switch here. It would be more readable and allow you to catch possible bugs , when some other VehicleType will be added.